scripts/bfasst/compare/waveform.py

```diff
@@ -73,536 +53,121 @@ def compare_netlists(self, design):
-        )
-        subprocess.run(["vvp", str(dsn)])
-        subprocess.run(["mv", "test.vcd", str(reversed_temp_vcd)])
-
-        # Setup gtkwave for the two files (creates file .gtkwaverc to set the zoom on the graphs to fit the window.)
-        gtkwave = Path(".gtkwaverc")
-        if(gtkwave.exists()):
-            gtkwave.unlink()
-        with gtkwave.open("x") as wavefile:
-            wavefile.write("do_initial_zoom_fit 1")
-        
-        commands = [
-            ["gtkwave", "-T", str(impl_tcl), "-o", str(impl_temp_vcd)],
-            ["gtkwave", "-T", str(reversed_tcl), "-o", str(reversed_temp_vcd)],
-        ]
-
-        # Procs is used to run these two wavefiles in parallel so both can be viewed against each other.
-        procs = [Popen(i) for i in commands]
-        for p in procs:
-            p.wait()
-
-        gtkwave.unlink()
-        # Finds how many lines are different in the two files.
-        dif = subprocess.getoutput(["diff -c " + str(impl_vcd) + " " + str(reversed_vcd)])
-        if diff_file.exists():
-            diff_file.unlink()
-        with diff_file.open("x") as file:
-            for line in dif:
-                file.write(line)
-
-        lines = 0
-        with diff_file.open() as file:
-            for line in file:
-                if len(line) != 0:
-                    lines = lines + 1
-
-        # If there are more than 32 lines different, the two designs must be unequivalent.
-        if lines > 32:
-            print("NOT EQUIVALENT! SEE " + str(parsed_diff_file) + " for more info")
-            parse_diff.parse_diff(impl_vcd, diff_file, parsed_diff_file)
-            if(parsed_diff_file.exists()):
-                with parsed_diff_file.open() as file:
-                    for line in file:
-                        print(line)
-            else:
-               subprocess.run(["diff", "-c", str(impl_vcd), str(reversed_vcd)]) 
-
-        else:
-            dsn.unlink()
-            diff_file.unlink()
-            is_equivalent = True
-        # removes unnecessary files.
-        impl_temp_vcd.unlink()
-        reversed_temp_vcd.unlink()
-        impl_fst.unlink()
-        reversed_fst.unlink()
-        lines = []
-        with impl_tcl.open("r") as fp:
-            lines = fp.readlines()
-        with impl_tcl.open("w") as fp:
-            for number, line in enumerate(lines):
-                if number not in [2,3]:
-                    fp.write(line)
-        with reversed_tcl.open("r") as fp:
-            lines = fp.readlines()
-        with reversed_tcl.open("w") as fp:
-            for number, line in enumerate(lines):
-                if number not in [2,3]:
-                    fp.write(line)
-        return is_equivalent
+    def run_test(self):
+        return parse_diff.check_diff(
+            paths
+        )  # Checks the two VCD files against each other. Returns either equivalent or not depending on how many lines are different.
 
     def check_compare_status(self, log_path):
```

Comment:
This function should have a default return value.

Reply:
Like adding default=CompareStatus.NOT_EQUIVALENT to the function? Out of curiosity, why? The function check_diff always returns true or false. Should I replace that to be true or default? Or should the default be added to run_test?

Reply:
Sorry, I was talking about check_compare_status.  Currently all the returns are inside if statements, and if none of them are true, it's going to return None, which is inconsistent with the rest of the branches.